scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -741,6 +742,48 @@ def small_sample_generator(n_dims):
                     hypotest(*samples, *args, axis=axis, **kwds)
 
 
+@pytest.mark.parametrize("axis", [0, 1])
+@pytest.mark.parametrize(("hypotest", "args", "kwds", "n_samples", "n_outputs",
+                          "paired", "unpacker"), axis_nan_policy_cases)
+def test_non_broadcastable(hypotest, args, kwds, n_samples, n_outputs, paired,
```

Comment:
This test checks for the correct error message in non-empty samples. It's a good test to have for all functions (not just paired-sample statistics), and there is an extra part for paired-sample statistics.